odml/tools/rdf_converter.py

```diff
@@ -0,0 +1,141 @@
+        else:
+            curr_node = node
+
+        self.g.add((curr_node, fmt.rdf_type(), URIRef(fmt.rdf_type())))
+
+        # adding doc to the hub
+        if isinstance(fmt, format.Document.__class__):
+            self.g.add((self.hub_root, odmlns.hasDocument, curr_node))
+
+        for k in fmt._rdf_map:
+            if k == 'id':
+                continue
+            elif (isinstance(fmt, format.Document.__class__) or
+                      isinstance(fmt, format.Section.__class__)) and k == "repository":
+                terminology_url = getattr(e, k)
+                if terminology_url is None or not terminology_url:
+                    continue
+                terminology_node = self._get_terminology_by_value(terminology_url)
+                if terminology_node:
+                    self.g.add((curr_node, fmt.rdf_map(k), terminology_node))
+                else:
+                    # adding terminology to the hub and to link with the doc
+                    node = URIRef(odmlns + str(uuid.uuid4()))
+                    self.g.add((node, odmlns.Terminology, URIRef(terminology_url)))
+                    self.g.add((self.hub_root, odmlns.hasTerminology, node))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+            # generating nodes for entities: sections, properties and bags of values
+            elif (isinstance(fmt, format.Document.__class__) or
+                      isinstance(fmt, format.Section.__class__)) and \
+                            k == 'sections' and len(getattr(e, k)) > 0:
+                sections = getattr(e, k)
+                for s in sections:
+                    node = URIRef(odmlns + str(s.id))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+                    self.save_element(s, node)
+            elif isinstance(fmt, format.Section.__class__) and \
+                            k == 'properties' and len(getattr(e, k)) > 0:
+                properties = getattr(e, k)
+                for p in properties:
+                    node = URIRef(odmlns + str(p.id))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+                    self.save_element(p, node)
+            elif isinstance(fmt, format.Property.__class__) and \
+                            k == 'value' and len(getattr(e, k)) > 0:
+                values = getattr(e, k)
+                bag = BNode(str(uuid.uuid4()))
+                self.g.add((curr_node, fmt.rdf_map(k), bag))
+                for v in values:
+                    self.g.add((bag, RDF.li, Literal(v)))
+            # adding entities' properties
+            else:
+                val = getattr(e, k)
+                if val is None or not val:
+                    continue
+                elif k == 'date':
+                    self.g.add((curr_node, fmt.rdf_map(k), Literal(val, datatype=XSD.date)))
+                self.g.add((curr_node, fmt.rdf_map(k), Literal(val)))
```

Comment:
Use an `else` clause here, otherwise a date will be added twice, once with the XSD datatype, once without.